### ISSUE TITLE
Get.testMode = true is ineffective

### ISSUE THREAD

**Issue by Presheda** [jonataslaw/getx#790]
unable to run navigation test as Get.testMode is ineffective 

`     test("test description", () async {
  
      SplashScreenController controller = SplashScreenController();
      await controller.fetchData();

    });

![Screenshot (115)](https://user-images.githubusercontent.com/26353504/99039009-d2f44000-2586-11eb-988a-2c860ef52b7e.png)

`

**Comment by jonataslaw** [comment]
Please provide a reproduction code

**Comment by Presheda** [comment]
Steps to reproduce 

**SplashScreenController**

``` import 'package:deggia_regular/Services/local_storage_service/storage_service.dart';
import 'package:deggia_regular/ui/route/route_names.dart';
import 'package:deggia_regular/ui/views/home/home_screen.dart';
import 'package:deggia_regular/utils/locator.dart';
import 'package:flutter/material.dart';
import 'package:get/get.dart';

class SplashScreenController extends GetxController {
  StorageService storageService = locator<StorageService>();

  @override
  void onInit() {
    super.onInit();
    fetchData();
  }

  Future fetchData() async {
    Get.testMode = true;
    bool isFirstTime = await storageService.firstTime();
    if (isFirstTime) {
      await storageService.setFirstTime();
      await Get.offAndToNamed(RouteName.on_boarding);
    } else {
      await Get.offAllNamed(RouteName.homepage);
    }
  }
}
````



```
group("first_time_load", () {
    setUp(() {
        var service = StorageServiceMock();

  when(service.firstTime()).thenAnswer((realInvocation) async => true);
  when(service.isLightMode()).thenAnswer((realInvocation) async => true);

  locator.registerFactory<StorageService>(() => service);
    });

 

    test("test description", () async {
      SplashScreenController controller = SplashScreenController();
      await controller.fetchData();
    });
}
)
```

**Comment by agr1277** [comment]
I'm also running into an issue with this. I am hoping this is a quick thing to fix. Here's what I found from the commits

`testMode` was added in [this commit](https://github.com/jonataslaw/getx/commit/b7fb4d99abb543ecb903aa339344312fc69c3928#diff-f5e0fcb66a5e988a730bde0db057055c0bbe94160e85e24377c1e30d8d3aa410R1035) 

The checking of this value appears to have been removed in [this commit](https://github.com/jonataslaw/getx/commit/85700051f39717d123d92a84aaae6aae6ff87f5c#diff-f5e0fcb66a5e988a730bde0db057055c0bbe94160e85e24377c1e30d8d3aa410R958)

If I'm reading this history correctly, checking `testMode` [here](https://github.com/jonataslaw/getx/blob/master/lib/get_navigation/src/extension_navigation.dart#L979) should resolve it? I'm unfamiliar with the justification behind the commits listed though, I could be misunderstanding how this feature is supposed to work now!
